Applications/Spire/Source/Ui/DropDownMenu2.cpp

```
@@ -0,0 +1,195 @@
+      m_selected_index(0),
+      m_max_displayed_items(5) {
+  setAttribute(Qt::WA_ShowWithoutActivating);
+  m_shadow = new DropShadow(true, false, this);
+  setObjectName("DropDownMenu");
+  setStyleSheet(R"(
+    #DropDownMenu {
+      background-color: #C8C8C8;
+    })");
+  auto layout = new QVBoxLayout(this);
+  layout->setContentsMargins(scale_width(1), 0, scale_width(1),
+    scale_height(1));
+  m_scroll_area = new ScrollArea(this);
+  m_scroll_area->setFocusProxy(parent);
+  m_scroll_area->setWidgetResizable(true);
+  layout->addWidget(m_scroll_area);
+  auto body = new QWidget(m_scroll_area);
+  m_list_layout = new QVBoxLayout(body);
+  m_list_layout->setContentsMargins({});
+  m_list_layout->setSpacing(0);
+  m_scroll_area->setWidget(body);
+  for(auto& item : items) {
+    m_list_layout->addWidget(item);
+    m_item_hovered_connections.AddConnection(item->connect_hovered_signal(
+      [=] { m_hovered_signal(item->get_value()); }));
+    m_item_selected_connections.AddConnection(item->connect_selected_signal(
+      [=] { on_item_selected(item->get_value(), item); }));
+  }
+  if(m_list_layout->count() > 0) {
+    update_height();
+    static_cast<DropDownMenuItem2*>(
+      m_list_layout->itemAt(0)->widget())->set_highlighted();
+    m_selected_index = 0;
+    m_current_index = 0;
+    m_selected_signal(get_value(*m_selected_index));
+    m_current_signal(get_value(*m_current_index));
+  }
+  hide();
+}
+
+connection DropDownMenu2::connect_current_signal(
+    const CurrentSignal::slot_type& slot) const {
+  return m_current_signal.connect(slot);
+}
+
+connection DropDownMenu2::connect_hovered_signal(
+    const HoveredSignal::slot_type& slot) const {
+  return m_hovered_signal.connect(slot);
+}
+
+connection DropDownMenu2::connect_selected_signal(
+    const SelectedSignal::slot_type& slot) const {
+  return m_selected_signal.connect(slot);
+}
+
+const QVariant& DropDownMenu2::get_value(int index) const {
+  if(m_list_layout->count() - 1 < index) {
```

See below about hiding errors.